src/main/java/org/opentripplanner/model/fare/FareProduct.java

```diff
@@ -0,0 +1,87 @@
+package org.opentripplanner.model.fare;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.opentripplanner.framework.lang.Sandbox;
+import org.opentripplanner.framework.tostring.ToStringBuilder;
+import org.opentripplanner.transit.model.basic.Money;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+
+/**
+ * A ticket that a user can purchase to travel.
+ * <p>
+ * It may be valid for the entirety of an itinerary or just for some of its legs.
+ *
+ * @param id       Identity for the
+ * @param name     Human-readable name of the product
+ * @param amount   Price
+ * @param duration Maximum duration of the product, if null then unlimited duration
+ * @param category Rider category, for example seniors or students
+ * @param medium   Medium to "hold" the fare, like "cash", "HSL app" or
+ */
+@Sandbox
+public record FareProduct(
+  FeedScopedId id,
+  String name,
+  Money amount,
+  @Nullable Duration duration,
```

Comment:
Usually a product have a price not an amount. Also, the `duration` is the type, but what is it? 

A good rule of thumbs is to us the template: "A (CLASS-NAME) has a (FIELD-NAME) and (FIELD-NAME) is part of (CLASS_NAME)". If this does not make any meaning, then you probably have not found good names.

Reply:
The GTFS spec calls the number (without currency) the "amount" and stuck with this. In the API it's called "price" so I'm happy renaming it in the internal model, too. See 6811254e077c7cd5ba76643ab70963a8e4805b56

Reply:
Duration is again a term from the [Fares V2 spec](https://docs.google.com/document/d/19j-f-wZ5C_kYXmkLBye1g42U-kvfSVgYLkkG5oyBauY/edit#bookmark=id.m6p95mdqpcid).

I agree that it's quite generic and could be renamed to something like "validityPeriod" or "validityDuration". It's currently not exposed in an API.

What is your suggestion?

Reply:
I ended up going with "validityDuration": 74b7d3898e9dee8d2f94691e514671809191111a

Reply:
I ended up going with "validityDuration": 74b7d3898e9dee8d2f94691e514671809191111a

Reply:
Let's rename this to `validity`.